.github/workflows/check-migrations.yml

```
@@ -0,0 +1,43 @@
+name: Check Migrations
+
+on:
+  pull_request:
+    paths:
+      # Trigger when files in components are updated
+      - "packages/visual-editor/src/components/**"
+
+jobs:
+  comment-on-pr:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get changed files
+        id: changes
+        run: |
+          # Get the list of files changed in this PR
+          git fetch origin ${{ github.event.pull_request.base.ref }}
+          git diff --name-only origin/${{ github.event.pull_request.base.ref }} > changed_files.txt
+
+          # Check for changes under components, excluding the migrations, testing, and registry subdirectories and .test. files
+          grep '^packages/visual-editor/src/components/' changed_files.txt | \
+            grep -vE '^packages/visual-editor/src/components/(migrations|testing|registry)' | \
+            grep -v -q '\.test\.' && echo "components_updated=true" >> $GITHUB_ENV || echo "components_updated=false" >> $GITHUB_ENV
+
+          grep -q '^packages/visual-editor/src/components/migrations/' changed_files.txt && echo "migrations_updated=true" >> $GITHUB_ENV || echo "migrations_updated=false" >> $GITHUB_ENV
+
+      - name: Comment on PR
+        if: env.components_updated == 'true' && env.migrations_updated == 'false'
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const prNumber = context.payload.pull_request.number;
+            const commentBody = "Warning: Component files have been updated but no migrations have been added.";
+
+            await github.rest.issues.createComment({
```

Can this fail the PR instead of just adding a warning comment? It might already do that and new actions just don't run in the PR they're added.

---

It could, but I don't think all component changes actually need migrations (like if it was a minor styling fix) so it really is more of a warning

---

Oooh when we have documentation, the action can also link to a place where u can see how to decide if u need to add a migration and how to actually add it

---

You could create an empty markdown file now and link to that in Github. We probably wouldn't want to link to the live doc site here.